magefile.go

```diff
@@ -0,0 +1,299 @@
+		if err := mage.VerifyDeps("v0.3.0", "", "", true); err != nil {
+			return err
+		}
+	*/
+
+	// TODO(mage): Support verifying go module.
+	// TODO(mage): Upstream should support setting minimum version.
+	/*
+		os.Setenv(envMinGoVersion, minGoVersion)
+		fmt.Println("Running go module linter...")
+		if err := mage.VerifyGoMod(scriptDir); err != nil {
+			return err
+		}
+	*/
+
+	fmt.Println("Running golangci-lint...")
+	if err := mage.RunGolangCILint(golangciVersion, false); err != nil {
+		return err
+	}
+
+	if err := Build(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Build runs go build
+func Build() error {
+	fmt.Println("Running go build...")
+
+	ldFlag, err := mage.GenerateLDFlags()
+	if err != nil {
+		return err
+	}
+
+	os.Setenv(envLDFLAGS, ldFlag)
+
+	if err := mage.VerifyBuild(scriptDir); err != nil {
+		return err
+	}
+
+	fmt.Println("Binaries available in the output directory.")
+	return nil
+}
+
+func BuildBinaries() error {
+	fmt.Println("Building binaries with goreleaser...")
+
+	ldFlag, err := mage.GenerateLDFlags()
+	if err != nil {
+		return err
+	}
+
+	os.Setenv(envLDFLAGS, ldFlag)
+
+	return sh.RunV(
+		"goreleaser",
+		"release",
+		"--clean",
+	)
+}
+
+func BuildBinariesSnapshot() error {
+	fmt.Println("Building binaries with goreleaser in snapshot mode...")
+
+	ldFlag, err := mage.GenerateLDFlags()
+	if err != nil {
+		return err
+	}
+
+	os.Setenv(envLDFLAGS, ldFlag)
+
+	return sh.RunV(
+		"goreleaser",
+		"release",
+		"--clean",
+		"--snapshot",
+		"--skip-sign",
+	)
+}
+
+// BuildImages build bom image using ko
+func BuildImages() error {
+	fmt.Println("Building images with ko...")
+
+	gitVersion := getVersion()
+	gitCommit := getCommit()
+	ldFlag, err := mage.GenerateLDFlags()
+	if err != nil {
+		return err
+	}
+	os.Setenv(envLDFLAGS, ldFlag)
+	os.Setenv("KOCACHE", "/tmp/ko")
+
+	if os.Getenv("KO_DOCKER_REPO") == "" {
+		return errors.New("missing KO_DOCKER_REPO environment variable")
+	}
+
+	return sh.RunV(
+		"ko",
+		"build",
+		"--bare",
+		"--platform=all",
+		"--tags", gitVersion,
+		"--tags", gitCommit,
+		moduleName,
+	)
+}
+
+// BuildImagesLocal build images locally and not push
+func BuildImagesLocal() error {
+	fmt.Println("Building image with ko for local test...")
+	if err := mage.EnsureKO(""); err != nil {
+		return err
+	}
+
+	ldFlag, err := mage.GenerateLDFlags()
+	if err != nil {
+		return err
+	}
+
+	os.Setenv(envLDFLAGS, ldFlag)
+	os.Setenv("KOCACHE", "/tmp/ko")
+
+	return sh.RunV(
+		"ko",
+		"build",
+		"--bare",
+		"--local",
+		"--platform=linux/amd64",
+		moduleName,
+	)
+}
+
+func BuildStaging() error {
+	fmt.Println("Ensuring mage is available...")
+	if err := pkg.EnsureMage(""); err != nil {
+		return err
+	}
+
+	if err := mage.EnsureKO(""); err != nil {
+		return err
+	}
+
+	if err := BuildImages(); err != nil {
+		return fmt.Errorf("building the images: %w", err)
+	}
+
+	return nil
+}
+
+func Clean() {
+	fmt.Println("Cleaning workspace...")
+	toClean := []string{"output"}
```

Comment:
As far as I understand we don't generate anything to a folder called `output` so this Clean func is not cleaning the repo properly. Instead we should remove `bin`, as we did in the Makefile.